src/tcp_src.c

```diff
@@ -94,6 +94,7 @@ static void tcp_connect_cb(uv_connect_t *req, int status) {
 static void resolve_cb(uv_getaddrinfo_t *req, int status, struct addrinfo *addr) {
     tcp_src_t *sl = req->data;
 
+    sl->resolve_req = NULL;
```

Comment:
shouldn't this be done after the NULL check on line 98?

Reply:
yes